solidity/test/TestMinimumStake.js

```diff
@@ -0,0 +1,75 @@
+import increaseTime, {duration} from './helpers/increaseTime';
+import latestTime from './helpers/latestTime';
+import {createSnapshot, restoreSnapshot} from "./helpers/snapshot"
+
+const BN = web3.utils.BN
+const chai = require('chai')
+chai.use(require('bn-chai')(BN))
+const expect = chai.expect
+
+const KeepToken = artifacts.require('./KeepToken.sol');
+const TokenStaking = artifacts.require('./TokenStaking.sol');
+const Registry = artifacts.require("./Registry.sol");
+
+contract('TokenStaking', function() {
+  let token, registry, stakingContract, minimumStakeBase, minimumStakeSteps, minimumStakeSchedule, keepDecimals;
+  const initializationPeriod = 10;
+  const undelegationPeriod = 30;
+
+  before(async () => {
+    token = await KeepToken.new();
+    registry = await Registry.new();
+    stakingContract = await TokenStaking.new(
+      token.address, registry.address, initializationPeriod, undelegationPeriod
+    );
+
+    minimumStakeBase = await stakingContract.minimumStakeBase();
+    minimumStakeSteps = await stakingContract.minimumStakeSteps();
+    minimumStakeSchedule = await stakingContract.minimumStakeSchedule();
+    keepDecimals = web3.utils.toBN(10).pow(web3.utils.toBN(18));
+  });
+
+  beforeEach(async () => {
+    await createSnapshot()
+  })
+
+  afterEach(async () => {
+    await restoreSnapshot()
+  })
+
+  describe("minimumStake", async () => {
+    it("returns max value when the schedule starts", async () => {
+      expect(await stakingContract.minimumStake()).to.eq.BN(
+        web3.utils.toBN(100000).mul(keepDecimals),
+        "Unexpected minimum stake amount"
+      );
+    })
+
+    it("returns max value right before the next schedule step", async () => {
+      let minimumStakeScheduleStart = await stakingContract.minimumStakeScheduleStart();
+      let timeForStepOne = minimumStakeScheduleStart.add(minimumStakeSchedule.div(minimumStakeSteps))
```

Comment:
What do you think about extracting `stepDuration` to a separate variable? I think it's easier to follow:
```
let minimumStakeScheduleStart = await stakingContract.minimumStakeScheduleStart();
let stepDuration = minimumStakeSchedule.div(minimumStakeSteps)
let timeForStepOne = minimumStakeScheduleStart.add(stepDuration)
```